src/main/java/com/google/sps/data/Recipe.java

```diff
@@ -8,12 +8,14 @@
   private ArrayList<String> ingredients;
   private ArrayList<String> steps;
   private ArrayList<String> tags;
+  private boolean published;
   private String authorID;
   private String description;
   private String imageBlobKey;
   private String name;
-  private int popularity;
+  private long popularity;
   private long id;
+  private long publicRecipeID;
```

Comment:
Now that you're storing the public recipe ID, I don't think you strictly need the published variable anymore. From what I understand from reading your code, if the recipe is not published, publicRecipeID = 0, and if it is published, publicRecipeID != 0. So you could replace your checks of the boolean published with those checks instead.

Reply:
that makes sense will do

Reply:
oh actually I do use the published variable to tell if the recipe entity is a PublicRecipe or a PrivateRecipe
bc if it's published and the public == 0 then it's a public recipe else it's a PrivateRecipe